analyses/focal-cn-file-preparation/rna-expression-validation.R

```diff
@@ -0,0 +1,166 @@
+# This script examines RNA-seq expression levels of genes that are called as
+# deletions.
+#
+# Chante Bethell for CCDL 2019
+#
+# #### USAGE
+# This script is intended to be run via the command line from the top directory
+# of the repository as follows:
+#
+# Rscript 'analyses/oncoprint-landscape/rna-expression-validation.R'
+
+#### Set Up --------------------------------------------------------------------
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+#### Directories and Files -----------------------------------------------------
+
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper sourcing of functions no
+# matter where this is called from
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Set path to results and plots directory
+results_dir <-
+  file.path(root_dir, "analyses", "focal-cn-file-preparation", "results")
+
+plots_dir <-
+  file.path(root_dir, "analyses", "focal-cn-file-preparation", "plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Read in data from tsv file (produced in `01-prepare-cn-file.R`)
+cn_df <- readr::read_tsv(file.path(results_dir, "annotated_cn.tsv"))
+
+# Read in RNA-seq expression data
+rsem_expression_polyA <-
+  readr::read_rds(file.path(root_dir,
+                            "data",
+                            "pbta-gene-expression-rsem-fpkm.polya.rds"))
+
+rsem_expression_stranded <-
+  readr::read_rds(file.path(
+    root_dir,
+    "data",
+    "pbta-gene-expression-rsem-fpkm.stranded.rds"
+  ))
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv"))
+
+#### Custom Function -----------------------------------------------------------
+
+merge_expression <-
+  function(copy_number_df, expression_df, metadata) {
+    # Given the focal copy number data.frame already annotated with the metadata,
+    # the RNA-seq expression data.frame, and the metadata, combine the data into one
+    # data.frame and plot expression values.
+    #
+    # Args:
+    #   copy_number_df: focal copy number data.frame annotated with the metadata
+    #   expression_df: RNA-seq expression data.frame
+    #   metadata: the relevant metadata data.frame
+    #
+    # Returns:
+    #   combined_df: data.frame with information from the focal CN, the
+    #                RNA-seq expression, and metadata data.frames
+    
+    # Change expression data.frame to long tidy format
+    long_expression <- expression_df %>%
```

Comment:
I'd be curious if switching the order of `melt` and `inner_join` here improved speed. Try joining the metadata first, then melting to long format. That would reduce the number of comparisons needed for the join. I'm not sure it will make a huge difference, as the big expense is presumably joining the two long data frames.

Reply:
I attempted this, but the code was a bit messier this way and it did not significantly improve speed so I left this section in the `melt` then `inner_join` order. I did however address all of your other comments above.